scripts/testing/README.md

```diff
@@ -0,0 +1,112 @@
+## Automated testing
+
+### Installation
+
+Atomated testing CLI depends on `gcloud` and `vault` CLIs so you need to install them. 
+
+#### Google cloud CLI
+Follow the [offical installation](https://cloud.google.com/sdk/docs/install) guide. Once installed, run `gcloud auth login` and authorize using SSO. 
+
+#### Vault
+Download and install the Hashicorp Vault client version 1.14.1 or newer or via brew with `brew install vault`
+1. `brew install vault`.
+2. [Generate](https://github.com/elastic/infra/tree/master/docs/vault#github-auth) a github token to authorize in Vault.
+3. Run `vault login --method github` to authorize your Vault cli.
+
+#### Connector service
+Run `make clean install` to generate executable files in `./bin` folder
+
+### Usage
+
+***Note:*** The working directory for `./bin/test-connectors` is `scripts/testing` so when you need to point to a file you should use it as a root directory. 
+e.g. `./bin/test-connectors run-test my-testing-environment-name --es-version 8.12-SNAPSHOT --test-case scenarios/clients/spo_full_and_incremental_syncs.yml`
+
+Run `./bin/test-connectors --help` or `./bin/test-connectors {command name} --help` to get more information about the cli. 
+
+#### Running test with Elastic cloud deployment
+If you want to run your test suite using a cloud Elasticsearch deployment follow the next steps: 
+1. Create a cloud deployment
+2. Download a credentials file (or create a new user)
+3. Run `./bin/test-connectors run-test my-testing-environment-name --es-host {host} --es-username {user name} --es-password {password} --test-case {path to the test case file}`
+
+#### Running test with local Elasticsearch
+If you want to run your tests with local Elasticsearch you need to specify `--es-version` option. Like
+
+`./bin/test-connectors run-test my-testing-environment-name --es-version 8.12-SNAPSHOT --test-case {path to the test case file}`
+
+In this case, the cli will deploy an Elasticsearch instance in the same VM where the connector service will be running. 
```

Comment:
VM here is VM in GCS?

Reply:
Is it possible to use local machine as a target?

Reply:
> VM here is VM in GCS?

Yes. 

> Is it possible to use local machine as a target?

no, only GCP is supported for now.